instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SupportabilityMetrics.java

```diff
@@ -100,44 +100,44 @@ private CounterNames() {}
 
   // this class is threadsafe.
   private static class KindCounters {
-    private final LongAdder server = new LongAdder();
-    private final LongAdder client = new LongAdder();
-    private final LongAdder internal = new LongAdder();
-    private final LongAdder consumer = new LongAdder();
-    private final LongAdder producer = new LongAdder();
+    private final AtomicLong server = new AtomicLong();
```

Comment:
I think `LongAdder` is important in normal code. We should have a wrapper around the two types based on classpath check, and can use `@IgnoreJreRequirement` to allow LongAdder

Reply:
`AtomicLong` counters seem acceptable here since they are only used when agent debug is enabled. Not sure whether we'll need our own high-throughput counters in instrumentation, since typically that kind of work will be delegated to the Metrics API.

Reply:
Ah thanks for pointing that out - agree if only when debug is enabled, AtomicLong is fine